설정 화면 구현

ji-yeon224 opened the pull request:
## #️⃣ Issue Number
- issue: #25 

## 🔎 작업 내용
- 설정 화면 UI구현
- 이용약관, 개인정보처리방침 웹뷰 연결
- 피드백 페이지 이동
- AlertView 수정
- 폰트 수정 
  - Body, Caption에 사용되는 폰트는 medium, regular여서 폰트 추가 및 수정 했습니다

## 📷 스크린샷
| 로그인 전 | 로그인 후 |
|:----:|:----:|
|<img src = "https://github.com/user-attachments/assets/1fba222a-74be-4256-9449-56ab7ebd43da" width = "300">| <img src = "https://github.com/user-attachments/assets/33504eac-9a81-42fe-a4c3-56eda31cb006" width = "300">|

|개인정보 처리방침|서비스 이용약관|
|:----:|:----:|
|<img src = "https://github.com/user-attachments/assets/cb129219-55c0-4ffc-afe5-2d6be922e084" width = "300">|<img src = "https://github.com/user-attachments/assets/90ad53a4-9ccd-40e8-9641-e2280a66c948" width = "300">|

|탈퇴 얼럿|로그아웃 얼럿|
|:----:|:----:|
<img src = "https://github.com/user-attachments/assets/3f52831a-8634-4ede-b447-f6d67c13cce5" width = "300">|<img src = "https://github.com/user-attachments/assets/a33dbbbf-ff0d-40f6-9a0a-c0334d6f4a27" width = "300">|



## 🛠️ 기타 공유 내용
- 로그인 관련, 버전정보는 아직 미구현입니다

ji-yeon224 committed:
feat: #25 설정 모듈 생성

ji-yeon224 committed:
feat: #25 SettingView 연결

ji-yeon224 committed:
feat: #25 NavigationBar title font 설정

ji-yeon224 committed:
feat: #25 회원 정보 뷰 구현

ji-yeon224 committed:
feat: #25 설정 아이템 리스트 구현

ji-yeon224 committed:
feat: #25 설정 리스트 뷰 구현

ji-yeon224 committed:
feat: #25 피드백 UI 구현

ji-yeon224 committed:
feat: #25 body, caption 폰트 에셋 추가 및 수정

ji-yeon224 committed:
feat: #25 정책 뷰 웹뷰로 연결

ji-yeon224 committed:
feat: #25 Alert 배경 추가 및 ui 수정

ji-yeon224 committed:
feat: #25 로그아웃 및 로그인 얼럿 추가

ji-yeon224 committed:
feat: #25 외부 연결 링크 Utility에 정의

ji-yeon224 committed:
feat: #25 피드백 화면 외부로 이동

ji-yeon224 committed:
feat: #25 로그인 화면 이동 이벤트 연결

ji-yeon224 committed:
feat: #25 SettingItemView 터치 영역 조절

usa4060 commented on Projects/Features/Setting/SettingFeatureInterface/Sources/SettingList/SettingItemListView.swift:
```diff
@@ -0,0 +1,45 @@
+      .padding(.top, .Number8)
+      .padding(.bottom, .Number4)
+      .padding(.horizontal, .Number16)
+      .frame(height: .Number33)
```
이거 figma상에서는 hug로 잡혀있어서 padding으로만 레이아웃 잡아야 할 것 같아요!
(.Number33 이라는 요상한 상수가 있길래 찾아옴 ㅋㅋ)

usa4060 commented on Projects/Features/Setting/SettingFeature/Sources/SettingReducer.swift:
```diff
@@ -0,0 +1,73 @@
+      case .feedBackTapped:
+        return .run { send in
+          if let url = ExternalURL.feedBack {
+            await openURL(url)
```
오,,, 내부 코드 보니까 `openURL`의 callBack을 `AsyncStream`으로 감싸서 `Concurrency`로 사용하게 해주는 친구인가보구만

usa4060 commented on Projects/Features/Setting/SettingFeature/Sources/SettingReducer.swift:
```diff
@@ -0,0 +1,73 @@
+      case .settingListTapped(.terms):
+        return .send(.delegate(.pushToPolicy(.terms)))
+      case .settingListTapped(.privacy):
+        return .send(.delegate(.pushToPolicy(.privacy)))
```
근데 이거 생각을 좀 해봤는데, .delegate이 액션의 경우 뎁스가 여러개의 navigation을 거슬러올라가게 되면
`.send(.delegate(.delegate(.delegate)))...` 이런식으로 delegate가 지칭하는 상위 Action이 무엇인지 직관적으로 알아보기가 좀 어려울 것 같은데 prefix로 상위 Action을 지칭할 수 있는 무언가를 추가하는건 어떨까요??

ji-yeon224 replied:
음.. 상위 Action을 지칭한다는게 뭔지 잘 모르겠는데, 하위 리듀서에서는 상위 리듀서가 뭔지 알 필요는 없다고 생각해요! 

SettingReducer에서 보내는 action은 상위 리듀서가 뭔지는 모르지만 `.pushToPolicy`를 처리해줘! 이거고, (.terms, .privacy는 액션이 아니라 상위 리듀서로 전달할 값임)
PIDAReducer에서 하위 액션을 처리할 때 `case let .setting(.delegate(.pushToPolicy(type))):` 이런 식으로 .setting에서 delegate한 `.pushToPolicy`액션을 처리한다~ 라고 한다면 상위 리듀서에서는 어디에서 온 액션인지 파악이 가능하지 않을까 싶습니다

그리고 PIDAReducer에서 Scope를 통해 정의되어있는 하위 리듀서의 액션만 처리할 수 있기 때문에 .delegate가 중첩으로 발생하는 경우는 없을 것 같아요!

usa4060 replied:
아 이해했습니다! 👍🏻

ji-yeon224 committed:
fix: #25 백그라운드 컬러 적용

usa4060 committed:
feat: #27 커스텀 NavigationBar 스와이프 pop 기능 구현

- UIViewControllerRepresentable을 활용하여 제스쳐 탐지 endable

usa4060 committed:
Merge branch 'feature/setting' of https://github.com/Team-PIDA/PIDA_iOS into feature/setting

# Conflicts:
#	Projects/Features/Setting/SettingFeatureInterface/Sources/Policy/PolicyView.swift

usa4060 reviewed (commented):
코멘트 확인 부탁드려영

ji-yeon224 committed:
fix: 상수값 수정

usa4060 committed:
Memory & Disk 관리용 TwoTierCache 구현 (#26)

* feat: #21 Cahce 모듈 생성

* feat: #21  Memory &  Disk Cache 저장소 생성 및 동작 구현

* chore: #21 Cahce 관련 protocol 인덴트 수정

* fix: #21 Cache의 범용적 사용을 위해, DefaultNamespace 이름 변경

- 해당 Cache관련 타입은 CacheNamespace를 준수하는 타입이기 때문에, 외부의 모듈에서 name이 겹쳐도 상관 x
- 따라서, 좀 더 범용적이고 직관적인 네이밍 사용하는 방식으로 수정

* fix: #21 Cache 컴포넌트 분류

* feat: #21 외부에서 사용하게 될 CacheActor 사용 관련 주석 추가 및 직관적인 변수명 수정

* fix: #21 DiskCacheStorage 사용 관련 주석 추가

* chore: #21 주석 정리

* chore: #21 merge conflict fix

usa4060 reviewed (approved):
확인이요~
